github/github_repo.go

```diff
@@ -66,6 +71,19 @@ func (repo *GithubRepo) oauthClient() *http.Client {
 	return oauth2.NewClient(context.Background(), tokenService)
 }
 
+func (repo *GithubRepo) githubClient() (*ghb.Client, error) {
+	oauthClient := repo.oauthClient()
+
+	if len(repo.baseURL) > 0 {
+		if len(repo.uploadURL) == 0 {
+			repo.uploadURL = repo.baseURL
+		}
```

Comment:
Would love to see all this condensed into a single function, something like `isGitHubEnterprise()`:
```
if repo.isGitHubEnterprise() {
  return ghb.NewEnterpriseClient(repo.baseURL, repo.baseURL, oauthClient)
}
return ghb.NewClient(oauthClient), nil
```